### PR TITLE
ParamsHelper addAttachment fix for Allure2 report generation

### PR DESCRIPTION
**Pull request by MeSaNei**
+Solution for file extension as a title may have unexpected consequences for Allure report generation. Can be the cause of Allure-Jenkins exceptions.

+ParamsHelper small enhancements
+Type class additions(file extensions)

### CODE REVIEW

File: src/main/java/ru/sbtqa/tag/allurehelper/Type.java [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
-    TARGZIP("application/x-gtar");
+    TEXT("text/plain", ".txt"),
+    XML("application/xml", ".xml"),
+    HTML("text/html", ".html"),
+    PNG("image/png", ".png"),
+    JPEG("image/jpeg", ".jpg"),
+    CSV("text/csv", ".csv"),
+    VIDEO("video/webm", ".webm"),
+    URI("text/uri-list", ".uri"),
+    TARGZIP("application/x-gtar", ".gtar");
```
Comment (clicman) [inline src/main/java/ru/sbtqa/tag/allurehelper/Type.java]: .tar.gz

Comment (MeSaNei) [reply src/main/java/ru/sbtqa/tag/allurehelper/Type.java]: I have used this resource for picking the right file file extension https://www.freeformatter.com/mime-types-list.html

File: src/main/java/ru/sbtqa/tag/allurehelper/ParamsHelper.java [hunk 2]
```diff
@@ -43,11 +45,36 @@ public static void addParam(String format, String[] parameters) {
     /**
      * Add attachment to allure report
      *
+     * @deprecated file extension as a title may have unexpected consequences by Allure report generation. Can be the cause
+     * of a Allure-Jenkins exceptions.
      * @param attachment as byte array.
      * @param title title of attachment. Shown at report as name of attachment
-     * @param type type of attachment
+     * @param type type of attachment     *
```
Comment (kosteman) [inline src/main/java/ru/sbtqa/tag/allurehelper/ParamsHelper.java]: remove this artifact

File: src/main/java/ru/sbtqa/tag/allurehelper/ParamsHelper.java [hunk 3]
```diff
@@ -3,10 +3,11 @@
 import io.qameta.allure.Allure;
 import io.qameta.allure.model.Status;
 import io.qameta.allure.model.StepResult;
-import static java.util.UUID.randomUUID;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static java.util.UUID.randomUUID;
```
Comment (kosteman) [inline src/main/java/ru/sbtqa/tag/allurehelper/ParamsHelper.java]: please do not shake imports

File: src/main/java/ru/sbtqa/tag/allurehelper/Type.java [hunk 4]
```diff
@@ -1,25 +1,37 @@
...
 
-    private Type(final String type) {
+    Type(final String type, final String extension) {
         this.type = type;
+        this.extension = extension;
     }
 
     @Override
     public String toString() {
         return type;
```
Comment (kosteman) [inline src/main/java/ru/sbtqa/tag/allurehelper/Type.java]: maybe now we also want to have an extension returning in toString method and individual getter for type

File: src/main/java/ru/sbtqa/tag/allurehelper/Type.java [hunk 5]
```diff
@@ -1,25 +1,37 @@
 package ru.sbtqa.tag.allurehelper;
 
+/**
+ * Enum for a attachment to addAttachment method in ParamsHelper class.
+ *
+ * @see ru.sbtqa.tag.allurehelper.ParamsHelper *
```
Comment (kosteman) [inline src/main/java/ru/sbtqa/tag/allurehelper/Type.java]: redundant *

**Review by kosteman — CHANGES_REQUESTED** [review]

**Review by kosteman — APPROVED** [review]

**Pull request merged**